tools/ReleaseNotes/data/whats-new-13.md

```diff
@@ -0,0 +1,1790 @@
+
+var api = builder.AddProject<Projects.Api>("api");
+
+var frontend = builder.AddNpmApp("frontend", "../frontend")
+    .WithPipelineConfiguration(context =>
+    {
+        // Get the build steps for this resource
+        var frontendBuild = context.GetSteps(context.Resource, WellKnownPipelineTags.BuildCompute);
+
+        // Get the build steps for the API resource
+        var apiBuild = context.GetSteps(api.Resource, WellKnownPipelineTags.BuildCompute);
+ 
+        // Make frontend build depend on API build
+        frontendBuild.DependsOn(apiBuild);
+    });
+
+await builder.Build().RunAsync();
+```
+
+The pipeline system includes:
+
+- **Global steps**: Define custom deployment steps with `builder.Pipeline.AddStep`
+- **Resource steps**: Resources contribute steps via `WithPipelineStepFactory`
+- **Dependency configuration**: Control step ordering with `WithPipelineConfiguration`
+- **Parallel execution**: Steps run concurrently when dependencies allow
+- **Built-in logging**: Use `context.Logger` to log step progress
+- **CLI execution**: Run specific steps with `aspire do <step-name>`
+
+#### Executing pipelines with aspire do
+
+Once you've defined your pipeline steps using the APIs above, you can execute them through the CLI using the new `aspire do` command. This command serves as the primary entry point for running pipeline steps, whether they're built-in steps like `build`, `publish`, and `deploy`, or custom steps you've defined in your AppHost.
+
+The `aspire do` command understands the entire pipeline graph, automatically resolving dependencies and executing steps in the correct order. For example, when you run `aspire do deploy`, it will automatically run any prerequisite steps (like `build` and `publish`) before executing the deployment itself.
+
+```bash
+# Execute a specific pipeline step (e.g., deploy)
+# This automatically runs all required steps: build → publish → deploy
+aspire do deploy
+
+# Execute with custom output path
+aspire do publish --output-path ./artifacts
+
+# Execute with specific environment
+aspire do deploy --environment Production
+
+# Execute with verbose logging
+aspire do deploy --log-level debug
+
+# Execute a custom step you defined (like the "validate" example above)
+aspire do validate
+```
+
+The `aspire do` command provides fine-grained control over deployment workflows, allowing you to execute any step in your pipeline independently while ensuring all dependencies are satisfied.
+
+For more details on the pipeline architecture, see [Deployment pipeline documentation](../deployment/pipeline-architecture.md).
+
+### Container Files as Build Artifacts
+
+Aspire 13 introduces a fundamental paradigm shift in how build artifacts are handled: **containers are now first-class build artifacts**, not just folders on a file system. This change enables reproducible, isolated, and portable builds that work consistently across development, CI/CD, and production environments.
```

Comment:
This section talks too much about "what are the benefits of using Docker to build your app". Instead I think it should focus more about the new feature - one resource can build a docker image with files in it. And another resource can "consume" or copy those files into its output.